src/rlx_prv_relup.erl

```diff
@@ -145,7 +155,14 @@ make_upfrom_script(State, Release, UpFrom) ->
                             "relup successfully created!"),
             {ok, State};
         {ok,_, Module,Warnings} ->
-            ?RLX_ERROR({relup_script_generation_warn, Module, Warnings});
+            case WarningsAsErrors of
```

Comment:
oh... that's unexpected indeed... this might be a bug. I will discuss it in the team if we should do anything about it. I guess you will have to decide for yourself what to do in the meantime, but a possible solution would be to do as your first implementation, but backup any old relup that might exists and restore it if it has be erroneously overwritten (or just delete the faulty generated relup if there was no file there already). Sorry for the hassle!!